migration-0.38/migration/dashboards.py

```diff
@@ -0,0 +1,113 @@
+                    # check if materialized column is there if yes then replace it
+                    materialized_name = updated_attribute[2][:-1] + "_" + updated_attribute[1].lower() + "_" + updated_attribute[0].replace('.', '_')
+                    if materialized_name in sql["query"]:
+                        updated_materialized_name =  updated_attribute[2][:-1] + "_" + updated_attribute[1].lower() + "_" + updated_attribute[0].replace('.', '$$')
+                        sql["query"] = sql["query"].replace(materialized_name, updated_materialized_name) 
+        elif query_type == "builder":
+            groupByNames = {}
+            for i  in range(len(widget['query']['builder']['queryData'])):
+                query_data = widget['query']['builder']['queryData'][i]
+                data_source = query_data.get('dataSource', 'No DataSource found')
+                if data_source != "logs":
+                    continue
+                print("Dashboard : {} , panel: {}, type: {}".format(data['title'], widget['title'], query_type))
+
+                # update aggregate attribute
+                query_data["aggregateAttribute"], updated = update_field(query_data["aggregateAttribute"], fields)
+                
+                # update filters
+                for j in range(0, len(query_data["filters"]["items"])):
+                    query_data["filters"]["items"][j]["key"], updated = update_field(query_data["filters"]["items"][j]["key"], fields)
+
+                # update group by 
+                for j in range(0, len(query_data["groupBy"])):
+                    oldKey = query_data["groupBy"][j]["key"]
+                    query_data["groupBy"][j], updated = update_field(query_data["groupBy"][j],fields)
+                    if updated:
+                        groupByNames[oldKey] = query_data["groupBy"][j]["key"]
+                
+                # update order by
+                for j in range(0, len(query_data["orderBy"])):
+                    if query_data["orderBy"][j]["columnName"] in groupByNames.keys():
+                        query_data["orderBy"][j]["columnName"] = groupByNames[query_data["orderBy"][j]["columnName"]]
+
+                # update the legends
+                for key, value in groupByNames.items():
+                    if r"{{" + key + r"}}" in query_data["legend"]:
+                        query_data["legend"] = query_data["legend"].replace(r"{{" + key + r"}}", r"{{" + value + r"}}")
+                # update the data
+                widget['query']['builder']['queryData'][i] = query_data
+
+            for i  in range(len(widget['query']['builder']['queryFormulas'])):
+                query_formula = widget['query']['builder']['queryFormulas'][i]
+                for key, value in groupByNames.items():
+                    if r"{{" + key + r"}}" in query_formula["legend"]:
+                        query_data["legend"] = query_formula["legend"].replace(r"{{" + key + r"}}", r"{{" + value + r"}}")
```

Comment:
Formula can have order by and having in dashboards.

Reply:
this is only for metrics